### PR TITLE
Issue 1226: Migration - Static Pages

### LINKED ISSUE #1226: Migration: Static Pages
**Description**

Migration of static pages to ReactJS.

- Terms & Conditions
- Privacy
- Cookies

**Tasks**
- [x] Create React component to wrap static page files
- [x] Create HTML files from pug files
- [x] Declare routes
- [x] Install HTML Webpack loader
- [x] TS configuration for HTML files

### PR DESCRIPTION
**Pull request by dleszcz**
this resolves #1226 

**Description**

Migration of static pages to ReactJS.

- Terms & Conditions
- Privacy
- Cookies

**Tasks**

- Create React component to wrap static page files
- Create HTML files from pug files
- Declare routes
- Install HTML Webpack loader
- TS configuration for HTML files

### CODE REVIEW

**Comment by simplypixi** [comment]
@carmenfan Could you review this PR?

File: frontend/routes/staticPage/staticPage.component.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import * as CookiesPageContent from './../../staticPages/cookies.html';
+import * as TermsPageContent from './../../staticPages/terms.html';
+import * as PrivacyPageContent from './../../staticPages/privacy.html';
+
+interface IProps {
+	match: any;
+}
+
+const staticPages = [
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply frontend/routes/staticPage/staticPage.component.tsx]: Yes. Definitely.

[the remark this answers is not in the document]
Comment (carmenfan) [reply frontend/routes/staticPage/staticPage.component.tsx]: Just to add, it basically as to work as before. The html pages should not be committed to the web (just send me a copy so I can test :) ) And those files should be defined in config.js. As onsite deployments or third party deployments of 3drepo.io would not be applicable to our in house generic T&Cs and we want to make it easy for support engineers to swap them for something meaningful.

**Comment by carmenfan** [comment]
Santiago had a look and he says it works/looks fine. But we'll need this updated with the dynamic version as discussed in the conversation above :)

File: frontend/routes/staticPage/staticPage.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dleszcz) [reply frontend/routes/staticPage/staticPage.component.tsx]: Ok, then files are defined in config file. I have one more question @carmenfan, should i delete these static html files from repository, right?

[the remark this answers is not in the document]
Comment (carmenfan) [reply frontend/routes/staticPage/staticPage.component.tsx]: yes

[the remark this answers is not in the document]
Comment (dleszcz) [reply frontend/routes/staticPage/staticPage.component.tsx]: This is done, but i changed legal pages paths in travis config file, and then the build is failed (catalog frontend/staticPages/legal/ exists). 
screenshot 2018-11-22 at 16 31 57
How can we deal with it?

**Comment by carmenfan** [comment]
@dleszcz  can you rebase/merge with staging? :)

File: frontend/routes/staticPage/staticPage.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (simplypixi) [reply frontend/routes/staticPage/staticPage.component.tsx]: @carmenfan Could you look at this?

**Comment by simplypixi** [comment]
@carmenfan Rebased ;) @dleszcz left one question above

File: frontend/routes/staticPage/staticPage.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (carmenfan) [reply frontend/routes/staticPage/staticPage.component.tsx]: This is trying to copy it from the test folder, which comes from the test repository.

Don't worry about this, i'll fix it when I test this branch :)

**Comment by carmenfan** [comment]
@dleszcz The config seems to assume the file location I give it is in staticPages/legal. can we change it to the root of the directory? Since it's a configuration setting it is a bit misleading. Initially I put my files in 3drepo.io/html/legal/* and got a blank screen. I only figured this out from the code.

**Comment by carmenfan** [comment]
@simplypixi 

1. Great :)
2. Sounds fair, but can we rename it to something like `resources` (or whatever is appropriate)? Looking at the pug folder there's some stuff other than legal pages in it, and some of them will be there for a while, and I think it's best if we merge the 2 folders (maybe not right now, but in the future!)
3. Sure, sounds good.

One thing about the travis test:
The submodule folder is from the test git repository, and this si where the legal test files are, so make sure the src matches this and it should pass :)
image

**Comment by simplypixi** [comment]
@carmenfan 

Yes, I can rename that directory. ;) 

And thank you for the tip!

**Comment by simplypixi** [comment]
@carmenfan Renamed ;)

**Comment by simplypixi** [comment]
@carmenfan Could you check what I did wrong in Travis config?

**Comment by carmenfan** [comment]
@simplypixi 

The test repo it was trying to pull did not have the files... I made a commit on this branch whilst I was reviewing it to fix this, but I guess it got lost somehow 😆 

Should hopefully fix it now.

**Comment by carmenfan** [comment]
@simplypixi  I assume this is ready for another look through?

**Comment by simplypixi** [comment]
> @simplypixi I assume this is ready for another look through?

Yes

**Comment by carmenfan** [comment]
@simplypixi 

I don't know if I'm doing somethign wrong but I'm seeing 2 banners on the static pages 😆 
image

The html file doesn't have a banner on it so i think it's the system generating that twice?
image

**Comment by simplypixi** [comment]
@carmenfan 
> @simplypixi
> 
> I don't know if I'm doing something wrong but I'm seeing 2 banners on the static pages 😆
> image
> 
> The html file doesn't have a banner on it so i think it's the system generating that twice?
> image

Whooah :D

Did you try to clean the browser cache?
Could you check `html` file path in request details? According to you second screenshot it should be something like that: 
image

If it's different, you have to change the file path in your client config. ;)

**Comment by carmenfan** [comment]
> 
> 
> @carmenfan
> 
> > @simplypixi
> > I don't know if I'm doing something wrong but I'm seeing 2 banners on the static pages 😆
> > image
> > The html file doesn't have a banner on it so i think it's the system generating that twice?
> > image
> 
> Whooah :D
> 
> Did you try to clean the browser cache?
> Could you check `html` file path in request details? According to you second screenshot it should be something like that:
> image
> 
> If it's different, you have to change the file path in your client config. ;)

mystery solved - I left out `legal` on the config.js 😆

**Pull request merged**